### ISSUE TITLE
Cloud Backup for User settings and Sessions ?

### ISSUE THREAD

**Issue by SyedMuhamadYasir** [brave/brave-browser#7532]
## Description 

The browser does not have any trivial method to implement cloud backup of user settings ( personal preferences, saved passwords ) and sessions ( active logins, site preferences )

Which makes it quite risky to depend on the browser in case if the machine or the hard drive, containing Brave, were to crash or fail.

Please implement such a cloud backup so the user's data is continuously synced and stays safe, agnostic of any device or platform's dependency.

The feature would closely resemble the 'sync' feature in Firefox and Chrome 

## Steps to Reproduce
None, its a feature that's lacking in the browser

## Expected result:

I would expect a cloud backup of session and user settings. similar to google chrome

## Reproduces how often: 

its a persistent feature

## Brave version (brave://version info)
Latest version

## Version/Channel Information:

- Can you reproduce this issue with the current release?
- Can you reproduce this issue with the beta channel?
- Can you reproduce this issue with the dev channel?
- Can you reproduce this issue with the nightly channel?

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields?
No
- Does the issue resolve itself when disabling Brave Rewards?
No
- Is the issue reproducible on the latest version of Chrome? 
Chrome doesn't have this issue

**Comment by walidmujahid** [comment]
This is a rather immediate comment, but I read this issue and I could not help but feel that 3Box could be the right direction. Or perhaps integrating IPFS -somehow in some way to some extent- into brave like Tor has been.

**Comment by srirambv** [comment]
cc: @rebron @bbondy @bsclifton @davidtemkin

**Comment by sidharthramesh** [comment]
I would like this feature as well. The sync chain code used for bookmarks can be used to encrypt all the settings (including passwords) and session data. This can be stored on the cloud as long as the sync chain code is secure. I would ditch chrome completely if this is implemented.

**Comment by intel352** [comment]
Agreed with @sidharthramesh. Browser-to-browser sync only works as long as you can depend on Brave to not encounter a bug and wipe out some or all data. As we've seen with the currently-disabled Sync, data loss is not unusual for this feature. There needs to be a cloud backup option, whether built-in to the browser (ideal) or else tell users where bookmark data is stored on their device so that they can ensure it is backed up to the cloud using a dedicated cloud storage client.
Any type of storage that has versioning would be best here, so that a rollback can occur if a data loss event were stored to the cloud.

**Comment by jsecretan** [comment]
So effectively, sync v2 does some of this.  But, for instance, it would not keep previous versions of records, which may be what you'd effectively want in a backup.  Will keep in the list.

**Comment by snvoid** [comment]
This is what is keeping me from switching over to Brave. Every time I try it this bothers me enough that I end up switching back to Chrome or Vivaldi.  So much of our data is already cloud based and backed up these days using strong encryption. Including password managers. Its frustrating that Brave doesn't offer something to users that lets us sync to a cloud
